File: moderate/column_names.pl
use strict;
use integer;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    $line--;
    my $r = chr( $line % 26 + ord 'A' );
    if ( $line >= 26 ) {
        $line = $line / 26 - 1;
        $r    = chr( $line % 26 + ord 'A' ) . $r;
        if ( $line >= 26 ) {
            $line = $line / 26 - 1;
            $r    = chr( $line % 26 + ord 'A' ) . $r;
        }
    }
    print "$r\n";
}
close(INFILE);

File: moderate/column_names.rb
def char(a)
  (a % 26 + 'A'.ord).chr
end

File.open(ARGV[0]).each_line do |a|
  a = a.to_i - 1
  d = char(a)
  if a >= 26
    a = a / 26 - 1
    d = char(a) + d
    if a >= 26
      a = a / 26 - 1
      d = char(a) + d
    end
  end
  puts d
end
